Src/FluentAssertions/Common/ExpressionExtensions.cs

```diff
@@ -87,29 +88,51 @@ public static MemberPath GetMemberPath<TDeclaringType, TPropertyType>(
                 case ExpressionType.Call:
                     var methodCallExpression = (MethodCallExpression)node;
 
-                    if (methodCallExpression is not { Method.Name: "get_Item", Arguments: [ConstantExpression argumentExpression] })
+                    if (methodCallExpression is not
+                        { Method.Name: "get_Item", Arguments: [ConstantExpression argumentExpression] })
                     {
                         throw new ArgumentException(GetUnsupportedExpressionMessage(expression.Body), nameof(expression));
                     }
 
                     node = methodCallExpression.Object;
-                    segments.Add($"[{argumentExpression.Value}]");
+                    singlePath = $"[{argumentExpression.Value}].{singlePath}";
+                    break;
+                case ExpressionType.New:
+                    var newExpression = (NewExpression)node;
+
+                    foreach (var member in newExpression.Arguments)
+                    {
+                        var expr = member.ToString();
+                        selectors.Add(expr[expr.IndexOf('.', StringComparison.Ordinal)..]);
+                        declaringTypes.Add(((MemberExpression)member).Member.DeclaringType);
+                    }
+
+                    node = null;
                     break;
 
                 default:
                     throw new ArgumentException(GetUnsupportedExpressionMessage(expression.Body), nameof(expression));
             }
         }
 
-        // If any members were accessed in the expression, the first one found is the last member.
         Type declaringType = declaringTypes.FirstOrDefault() ?? typeof(TDeclaringType);
 
-        IEnumerable<string> reversedSegments = segments.AsEnumerable().Reverse();
-        string segmentPath = string.Join(".", reversedSegments);
+        if (singlePath is null)
```

Comment:
❓ What happens if `singlePath` is non-null and `selectors` isn't empty?

Reply:
As I can see, this cannot happen, because you either have: `.Excluding(p => p.NestedField)`(which results in `singlePath` to be non-`null` and `segments` to be empty) or `.Excluding(p => new { p.NestedField } )` (which results in `singlePath` null + `segments` not empty).

Reply:
My point was that the code doesn't make that clear.

Reply:
Should I add a comment for this?

Reply:
Actually, we have do do it exactly like this, because in case of `.Excluding(p => new { })` `singlePath` stays `null` and the `segments` stays empty.

Reply:
In that case, I would make it an `if`, `else if` and `else`. Even though the huge `switch` is difficult to process, you at least can see what will happen at the end.